agent/Cargo.toml

```diff
@@ -13,38 +13,38 @@ akri-onvif = { path = "../discovery-handlers/onvif", optional = true }
 akri-opcua = { path = "../discovery-handlers/opcua", optional = true }
 akri-udev = { path = "../discovery-handlers/udev", optional = true }
 akri-shared = { path = "../shared" }
-anyhow = "1.0.38"
-async-stream = "0.3"
-async-trait = "0.1.0"
-blake2 = "0.9.0"
-chrono = "0.4.10"
+anyhow = "1.0.43"
+async-stream = "0.3.2"
+async-trait = "0.1.51"
+blake2 = "0.9.2"
+chrono = "0.4.19"
 cfg-if = "1.0.0"
 env_logger = "0.9.0"
-futures = { version = "0.3.1", package = "futures" }
-futures-core = "0.3"
-futures-util = "0.3"
-hyper = "0.14.2"
-h2 = "=0.3.3"
-kube = { version = "0.59.0",  features = ["derive"] }
+futures = { version = "0.3.17", package = "futures" }
+futures-core = "0.3.17"
+futures-util = "0.3.17"
+hyper = "0.14.12"
+h2 = "0.3.3"
```

Comment:
What do we do about making sure `h2` is fixed? this removes that

Reply:
And more generally, would this mess with other constraints? Say if we needed at least or at most version x rather than just  = x.y.z?

Reply:
> What do we do about making sure `h2` is fixed to `0.3.3`? this removes the preceding `=`

Good catch.
Looks like there's an issue already capturing this https://github.com/killercup/cargo-edit/issues/454 

Options: 
1. revert this toml files update change and keep update dependencies as is only in the Cargo.lock
2. modify the upgrade command to exclude the toml file with the h2 dependency.
3. wait for the cargo upgrade issue to be fixed (or contribute to fix it).

Maybe do 1 & 3

Reply:
Yeah #2 seems tricky. That issue has been up for a while so contributing the fix seems like the best bet. Would you be willing to take it on?